### ISSUE TITLE
Add `balance` flag to `random_split`

### ISSUE THREAD

**Issue by MohamedAliRashad** [pytorch/pytorch#75582]
### 🚀 The feature, motivation and pitch

It will be much easier to add a balance flag tp `random_split` for balancing between classes and not needing to go to `Sampler` or `train_test_split` from `sklearn`.

### Alternatives

`Sampler` or `train_test_split` from `sklearn`

### Additional context

Just make it for me, please ^^

**Comment by MohamedAliRashad** [comment]
@ejguan 
splits train and test data between classes in balance ... takes from every class the same number of samples tp build the splits.
If it's not possible the function can raise an error

**Comment by ejguan** [comment]
Then, why not just use `random_split(ds, [len(ds)/2, len(ds)/2])`

**Comment by MohamedAliRashad** [comment]
No, i don't mean balancing the train and test splits. I mean building the train split from equal number of samples between all classes so if i have 5 classes and there is no balance between them the algorithm makes sure that the sampling is balanced among them (2 samples from each class for example)

**Comment by ejguan** [comment]
I am confused... Then, how `train_test_split` from `sklearn` is going to help you in this case.

If you want to make sure DataLoader yields data with equal number of samples per class, I would suggest you to make your own sampler.

**Comment by MohamedAliRashad** [comment]
@ejguan 
`train_test_split` has an argument named `stratify` that does the balancing trick. (I don't find its name intuitive and would much appreciate the term `class_balance`

**Comment by ejguan** [comment]
Okay, I get it. So you want to stratify splits the dataset. This sounds like a reasonable feature request.
Feel free to open a PR to add such feature on `random_split`.
